client/src/relatedProducts/RelatedProductsView/RelatedProducts.jsx

```diff
@@ -27,55 +27,37 @@ export default class RelatedProducts extends React.Component {
     this.handleAddToOutfit = this.handleAddToOutfit.bind(this);
     this.getRelatedStateData = this.getRelatedStateData.bind(this);
     this.getOutfitData = this.getOutfitData.bind(this);
-
+    this.product = this.product.bind(this)
+    this.style = this.style.bind(this)
+    this.outFit = this.outFit.bind(this)
   }
 
   componentDidMount () {
     //(products request)
-    let getProduct = new Promise((resolve, reject) => {
-      let result=[]
-      this.props.state.relatedProducts.forEach((productId) => {
-        return productsWithId(productId)
-          .then(data => {
-            result.push(data.data)
-            if (result.length === this.props.state.relatedProducts.length) {
+    let product = this.product()
```

Comment:
So slick breaking all these functions into methods that can be called. In my current local, I implemented them for update as well. Makes everything cleaan

Reply:
awesome - used this comment to go ahead and approve the merge - since your in agreement with changes that affected your components code 🔥 - thanks for being open to the refactor - teamwork makes the dreamwork